src/hazard/sources/osc_zarr.py

```diff
@@ -149,9 +149,16 @@ def write(
         spatial_coords: Optional[bool] = True,
     ):
         if self.write_xarray_compatible_zarr and spatial_coords:
-            self.write_data_array(f"{path}_xarray", da)
-
-        self.write_zarr(path, da, chunks)
+            pp = PurePosixPath(path)
+            if da.name != pp.name:
```

Comment:
if this is a requirement, perhaps move this logic into `write_data_array` instead:

```python
write_data_array(path, da, chunks):
    pp = PurePosixPath(path)
    da = da.copy()
    da.name = pp.name
    path = pp.parent
    ...
```